commit/chainfee/outcome.go

```diff
@@ -101,8 +107,53 @@ func (p *processor) Outcome(
 		"gasPrices", gasPrices,
 		"consensusTimestamp", consensusObs.TimestampNow,
 	)
-	out := Outcome{GasPrices: gasPrices}
-	return out, nil
+
+	inflightChainFeeUpdates := make(map[cciptypes.ChainSelector]time.Time)
+	for _, gasPriceUpdate := range gasPrices {
+		chainSel := gasPriceUpdate.ChainSel
+		inflightChainFeeUpdates[chainSel] = time.Time{}
+		oldChainFeeUpdate, ok := consensusObs.ChainFeeUpdates[chainSel]
+		if ok {
+			inflightChainFeeUpdates[chainSel] = oldChainFeeUpdate.Timestamp
+		}
+	}
+
+	return newPricesOutcome(gasPrices, inflightChainFeeUpdates, int64(p.cfg.InflightPriceCheckRetries)), nil
+}
+
+// computeInflightPricesOutcome is called if in this round we wait for prices to appear OnChain.
+// If we still wait for some prices it will decrement the number of available retries.
+// If all prices appeared OnChain or no retries left it sends an empty outcome so we can transmit fresh prices in the
+// next round.
+func (p *processor) computeInflightPricesOutcome(
+	lggr logger.Logger, consensusObs Observation, prevOutcome Outcome,
+) Outcome {
+	lggr.Debugw("checking for previously transmitted chain fee price updates to appear on-chain",
```

Comment:
I think this can be Infow also, it won't get spammed, just once per round thing if at all.